pkg/fipcontroller/controller.go

```diff
@@ -41,77 +42,87 @@ func NewController(config *Configuration) (*Controller, error) {
 		HetznerClient:    hetznerClient,
 		KubernetesClient: kubernetesClient,
 		Configuration:    config,
-		NodeName:         os.Getenv("NODE_NAME"),
 	}, nil
 }
 
 func NewControllerConfiguration() (*Configuration, error) {
 	var config Configuration
 
+	// Read config from file if present
 	file, err := ioutil.ReadFile("config/config.json")
-	if err != nil {
-		return nil, fmt.Errorf("could not open config file: %v", err)
+	if err == nil {
+		err = json.Unmarshal(file, &config)
+		if err != nil {
+			return nil, fmt.Errorf("could not decode config: %v", err)
+		}
 	}
 
-	err = json.Unmarshal(file, &config)
-	if err != nil {
-		return nil, fmt.Errorf("could not decode config: %v", err)
+	// overwrite existing configs from ENV vars if present
+	if apiToken := os.Getenv("HETZNER_CLOUD_API_TOKEN"); apiToken != "" {
+		config.HetznerAPIToken = apiToken
+	}
+	if floatingIP := os.Getenv("HETZNER_CLOUD_FLOATING_IP"); floatingIP != "" {
+		config.FloatingIPAddress = floatingIP
+	}
+	if nodeName := os.Getenv("KUBERNETES_NODE_NAME"); nodeName != "" {
+		config.NodeName = nodeName
+	}
+	if nodeAddressType := os.Getenv("KUBERNETES_NODE_ADDRESS_TYPE"); nodeAddressType != "" {
```

Comment:
This should be a flag.

Reply:
Should be resolved with https://github.com/cbeneke/hcloud-fip-controller/pull/13/commits/c291e62932140c4237a3ebad0eb7149c9c8586db